static/js/components/ViewProfileDisplay.js

```diff
@@ -0,0 +1,210 @@
+/* global SETTINGS: false */
+import React from "react"
+import { compose } from "redux"
+import { connect } from "react-redux"
+import { connectRequest } from "redux-query-react"
+import { find, fromPairs, propEq } from "ramda"
+import { createStructuredSelector } from "reselect"
+import { MetaTags } from "react-meta-tags"
+
+import {
+  EMPLOYMENT_EXPERIENCE,
+  EMPLOYMENT_SIZE,
+  GENDER_CHOICES,
+  VIEW_PROFILE_PAGE_TITLE
+} from "../constants"
+import queries from "../lib/queries"
+import { currentUserSelector } from "../lib/queries/users"
+import { formatTitle } from "../util/util"
+
+import type { Country, CurrentUser } from "../flow/authTypes"
+import { setDrawerState } from "../reducers/drawer"
+
+type StateProps = {|
+  currentUser: ?CurrentUser,
+  countries: Array<Country>
+|}
+
+type DispatchProps = {
+  updateDrawer(state): void
+}
+
+type Props = {|
+  ...StateProps,
+  ...DispatchProps
+|}
+
+export function ViewProfileDisplay(props: Props) {
+  const { currentUser, countries, updateDrawer } = props
+
+  return countries && currentUser ? (
+    <div className="container auth-page registration-page">
+      <MetaTags>
+        <title>{formatTitle(VIEW_PROFILE_PAGE_TITLE)}</title>
+      </MetaTags>
+      <div className="auth-header row">
+        <h1 className="col-6">Profile</h1>
+        <div className="col-6 profile-button-col">
+          <button
+            type="submit"
+            onClick={() => {
+              updateDrawer("profileEdit")
+            }}
+            className="btn btn-danger profile-btn"
+          >
+            Edit
+          </button>
+        </div>
+      </div>
+      <div className="auth-card card-shadow row">
+        <div className="container profile-container ">
+          <div className="row">
+            {currentUser.is_authenticated ? (
+              <div className="col-12 auth-form">
+                <div className="row profile-row">
+                  <div className="col profile-label">First Name</div>
+                  <div className="col">
+                    {currentUser.legal_address.first_name}
+                  </div>
+                </div>
+                <div className="row profile-row">
+                  <div className="col profile-label">Last Name</div>
+                  <div className="col">
+                    {currentUser.legal_address.last_name}
+                  </div>
+                </div>
+                <div className="row profile-row">
+                  <div className="col profile-label">Full Name</div>
+                  <div className="col">{currentUser.profile.name}</div>
+                </div>
+                {currentUser.legal_address.street_address.map((line, idx) => (
+                  <div className="row profile-row" key={idx}>
+                    <div className="col profile-label">Address {idx + 1}</div>
```

Comment:
Not sure if this matters but in the design it is "Address" then "Address 2". There is no "Address 1"

Reply:
If it's not too hard, can we hide "address2" by default, and only display it when user clicks to add another line? Otherwise it is suggesting to some users that a 2nd line is required instead of optional.

Reply:
done